app/src/main/java/to/bitkit/viewmodels/TransferViewModel.kt

```diff
@@ -60,10 +68,93 @@ class TransferViewModel @Inject constructor(
+    var retryTimes = 0
+
     // region Spending
 
-    fun onOrderCreated(order: IBtOrder) {
-        _spendingUiState.update { it.copy(order = order, isAdvanced = false, defaultOrder = null) }
+    fun onClickMaxAmount() {
+        _spendingUiState.update {
+            it.copy(
+                satsAmount = it.maxAllowedToSend,
+                overrideSats = it.maxAllowedToSend,
+            )
+        }
+        updateLimits(false)
+    }
+
+    fun onClickQuarter() {
+        val quarter = (_spendingUiState.value.balanceAfterFee.toDouble() * QUARTER).roundToLong()
+
+        if (quarter > _spendingUiState.value.maxAllowedToSend) {
+            setTransferEffect(
+                TransferEffect.ToastError(
+                    title = context.getString(R.string.lightning__spending_amount__error_max__title),
+                    description = context.getString(
+                        R.string.lightning__spending_amount__error_max__description
+                    ).replace("{amount}", _spendingUiState.value.maxAllowedToSend.toString()),
+                )
+            )
+        }
+
+        _spendingUiState.update {
+            it.copy(
+                satsAmount = min(quarter, it.maxAllowedToSend),
+                overrideSats = min(quarter, it.maxAllowedToSend),
+            )
+        }
+        updateLimits(false)
+    }
+
+    fun onConfirmAmount() {
+        if (_transferValues.value.maxLspBalance == 0uL) {
+            setTransferEffect(
+                TransferEffect.ToastError(
+                    title = context.getString(R.string.lightning__spending_amount__error_max__title),
+                    description = context.getString(
+                        R.string.lightning__spending_amount__error_max__description_zero
+                    ),
+                )
+            )
+            return
+        }
+
+        _spendingUiState.update { it.copy(isLoading = true) }
+
+        viewModelScope.launch {
+            blocktankRepo.createOrder(_spendingUiState.value.satsAmount.toULong())
+                .onSuccess { order ->
+                    onOrderCreated(order)
+                    _spendingUiState.update { it.copy(isLoading = false) }
+                }.onFailure { e ->
+                    setTransferEffect(TransferEffect.ToastException(e))
+                    _spendingUiState.update { it.copy(isLoading = false) }
+                }
+        }
+    }
+
+    fun onAmountChanged(sats: Long) {
+        if (sats > _spendingUiState.value.maxAllowedToSend) {
+            setTransferEffect(
+                TransferEffect.ToastError(
+                    title = context.getString(R.string.lightning__spending_amount__error_max__title),
+                    description = context.getString(
+                        R.string.lightning__spending_amount__error_max__description
+                    ).replace("{amount}", _spendingUiState.value.maxAllowedToSend.toString()),
+                )
+            )
+        }
```

Comment:
in the RN app, the amount will not update if it would exceed the max limit.

I think we can easily do the same here with this code right after the effect:
```
_spendingUiState.update { it.copy(overrideSats = it.satsAmount) }
return
```